### ISSUE TITLE
Sort memory table by offset address

### ISSUE THREAD

**Issue by MCUdude** [avrdudes/avrdude#1606]
@stefanrueger I remember you mentioning creating a PR that would sort the memory table that gets printed when the `part` command is executed. Is this something that should be done before the 7.3 release?

```
$ ./avrdude -cxplainedpro_updi -pattiny817 -t
avrdude: AVR device initialized and ready to accept instructions
avrdude: device signature = 0x1e9320 (probably t817)

avrdude: processing -t interactive terminal
avrdude> part
ATtiny817 with programming modes UPDI, SPM

Memory           Size  Pg size  Offset
--------------------------------------
eeprom            128       32  0x1400 
flash            8192       64  0x8000 
fuses              10        1  0x1280 
fuse0/wdtcfg        1        1  0x1280 
fuse1/bodcfg        1        1  0x1281 
fuse2/osccfg        1        1  0x1282 
fuse4/tcd0cfg       1        1  0x1284 
fuse5/syscfg0       1        1  0x1285 
fuse6/syscfg1       1        1  0x1286 
fuse7/append        1        1  0x1287 
fuse8/bootend       1        1  0x1288 
lock                1        1  0x128a 
prodsig/sigrow     64       64  0x1100 
signature           3        1  0x1100 
sernum             10        1  0x1103 
osccal16            2        1  0x1118 
osccal20            2        1  0x111a 
tempsense           2        1  0x1120 
osc16err            2        1  0x1122 
osc20err            2        1  0x1124 
userrow/usersig    32       32  0x1300 
io               4352        1       0 
sram              512        1  0x3e00 
sib                32        1       0 

Variants         Package  F max   T range         V range       
----------------------------------------------------------------
ATtiny817-MF     VQFN24   20 MHz  [-40 C, 125 C]  [1.8 V, 5.5 V]
ATtiny817-MFR    VQFN24   20 MHz  [-40 C, 125 C]  [1.8 V, 5.5 V]
ATtiny817-MN     VQFN24   20 MHz  [-40 C, 105 C]  [1.8 V, 5.5 V]
ATtiny817-MNR    VQFN24   20 MHz  [-40 C, 105 C]  [1.8 V, 5.5 V]
ATtiny817-MNRES  VQFN24   20 MHz  [-40 C, 105 C]  [1.8 V, 5.5 V]

```

**Comment by stefanrueger** [comment]
That's already happened in the following sense: In the current git main, `part` *always* lists memories in canonical order defined by memory group order first (eeprom, flash, fuses, lock, sigrow, ...), *then* by offset within each memory group. Here the code: https://github.com/avrdudes/avrdude/blob/03d786e8b01e121988c42e83e5b87be3dfde6e34/src/avr.c#L1562-L1592

I like this sorting better than just ordering by offset. For example if a user looks for fuse memories in the list they can be sure that they come after eeprom and flash. If it was by offset only the user would have to wonder first how Microchip decided to layout the memories this time round for that part.

TLDR; Memory order is no longer at mercy of reshuffling through inheritance of parts and deletion/insertion of memories.
